tests/test_conf.py

```diff
@@ -0,0 +1,293 @@
+        self.conf.save_config(c)
+        c = self.conf.open_config()
+        self.assertEqual(self.sections + ['testsection'], c.sections())
+        c.remove_section('testsection')
+        self.conf.save_config(c)
+
+    @ordered
+    def test_get_oauth(self):
+        """
+        Testing get_oauth()
+        """
+        oauth = self.conf.get_oauth()
+        self.assertEqual(oauth['access_token'], 'f6952d6eef555ddd87aca66e56b91530222d6e318414816f3ba7cf5bf694bf0f')
+        self.assertEqual(oauth['token_type'], 'Bearer')
+        self.assertEqual(oauth['expires_in'], '3600')
+        self.assertEqual(oauth['scope'], 'user-modify-playback-state ugc-image-upload user-library-modify')
+        self.assertEqual(oauth['expires_at'], '15848754832')
+        self.assertEqual(oauth['refresh_token'], '737dd1bca21d67a7c158ed425276b04581e3c2b1f209e25a7cff37d8cb333f0f')
+        self.conf.CONFIG_FILE = 'empty.conf'
+        oauth = self.conf.get_oauth()
+        self.assertEqual(oauth, {})
+
+    @ordered
+    def test_get_blacklist(self):
+        """
+        Testing get_blacklist()
+        """
+        blacklist = self.conf.get_blacklist()
+        self.assertEqual(blacklist['tracks'], {})
+        self.assertEqual(blacklist['artists'], {})
+        self.conf.CONFIG_FILE = 'empty.conf'
+        blacklist = self.conf.get_blacklist()
+        self.assertEqual(blacklist['tracks'], {})
+        self.assertEqual(blacklist['artists'], {})
+
+    @ordered
+    def test_add_to_blacklist(self):
+        """
+        Testing add_to_blacklist()
+        """
+        test_artist = {'name': 'frankie', 'uri': 'spotify:artist:testuri0frankie'}
+        test_track = {'name': 'nights', 'uri': 'spotify:track:testuri0nights', 'artists': [{'name': 'frankie'}]}
+        self.conf.add_to_blacklist(test_artist, test_artist['uri'])
+        self.conf.add_to_blacklist(test_track, test_track['uri'])
+        blacklist = self.conf.get_blacklist()
+        test_track['artists'] = ['frankie']
+        self.assertEqual(blacklist['tracks']['spotify:track:testuri0nights'], test_track)
+        self.assertEqual(blacklist['artists']['spotify:artist:testuri0frankie'], test_artist)
+
+    @ordered
+    def test_remove_from_blacklist(self):
+        """
+        Testing remove_from_blacklist()
+        """
+        # ensure proper functionality
+        artist_uri = 'spotify:artist:testuri0frankie'
+        track_uri = 'spotify:track:testuri0nights'
+        self.conf.remove_from_blacklist(artist_uri)
+        self.conf.remove_from_blacklist(track_uri)
+        blacklist = self.conf.get_blacklist()
+        self.assertEqual(blacklist['tracks'], {})
+        self.assertEqual(blacklist['artists'], {})
+
+        # ensure faulty uri returns None
+        res = self.conf.remove_from_blacklist('this-is-not-a-uri')
+        self.assertEqual(res, None)
+
+        # coverage lol
+        self.conf.remove_from_blacklist('spotify:track:thisdoesnotexist')
```

Comment:
But why? Because of the difference in schema between `'this-is-not-a-uri'` and `'spotify:track:thisdoesnotexist'`?